src/osp/Resource/load_tinygltf.cpp

```diff
@@ -0,0 +1,284 @@
+    }
+
+    // Store textures
+    rImportData.m_textures.resize(rImporter.textureCount());
+    for (UnsignedInt i = 0; i < rImporter.textureCount(); i ++)
+    {
+        Optional<TextureData> tex = rImporter.texture(i);
+
+        if ( ! bool(tex) )
+        {
+            continue;
+        }
+
+        // Create and keep track of resource Id
+        ResId const texRes = rResources.create(gc_texture, pkg, format_name(rImporter.textureName(i), i));
+        rImportData.m_textures[i] = rResources.owner_create(gc_texture, texRes);
+
+        // Add data to resource
+        rResources.data_add<TextureData>(gc_texture, texRes, std::move(*tex));
+
+        // Keep track of which image this texture uses
+        if (ResIdOwner_t const& imgRes = rImportData.m_images.at(tex->image());
+            imgRes.has_value())
+        {
+            ResIdOwner_t imgOwner = rResources.owner_create(gc_image, imgRes);
+            rResources.data_add<TextureImgSource>(gc_texture, texRes, TextureImgSource{std::move(imgOwner)} );
+        }
+    }
+
+    // Store meshes
+    rImportData.m_meshes.resize(rImporter.meshCount());
+    for (UnsignedInt i = 0; i < rImporter.meshCount(); i ++)
+    {
+        Optional<MeshData> mesh = rImporter.mesh(i);
+
+        if ( ! bool(mesh) )
+        {
+            continue;
+        }
+
+        ResId const meshRes = rResources.create(gc_mesh, pkg, format_name(rImporter.meshName(i), i));
+        rResources.data_add<MeshData>(gc_mesh, meshRes, std::move(*mesh));
+        rImportData.m_meshes[i] = rResources.owner_create(gc_mesh, meshRes);
+    }
+
+    // Store materials
+    rImportData.m_materials.reserve(rImporter.materialCount());
+    for (UnsignedInt i = 0; i < rImporter.materialCount(); i ++)
+    {
+        rImportData.m_materials.emplace_back(rImporter.material(i));
+    }
+
+    // Store various node data
+
+    auto &rNodeExtras = rResources.data_add<TinyGltfNodeExtras_t>(
+                                restypes::gc_importer, res);
+
+    UnsignedInt parentsSet = 0;
+
+    UnsignedInt const objCount = rImporter.object3DCount();
+    rNodeExtras                 .resize(objCount);
+    rImportData.m_objNames      .resize(objCount);
+    rImportData.m_objMeshes     .resize(objCount);
+    rImportData.m_objMaterials  .resize(objCount);
+    rImportData.m_objTransforms .resize(objCount);
+    rImportData.m_objParents    .resize(objCount);
+    rImportData.m_objChildren   .ids_reserve(objCount);
+    rImportData.m_objChildren   .data_reserve(objCount);
+
+    for (UnsignedInt i = 0; i < objCount; i ++)
+    {
+        Pointer<ObjectData3D const> obj         = rImporter.object3D(i);
+        int const instance                      = obj->instance();
+        ObjectInstanceType3D const instanceType = obj->instanceType();
+
+        if ( obj == nullptr )
+        {
+            continue;
+        }
+
+        rImportData.m_objNames[i] = rImporter.object3DName(i);
```

Comment:
Code scanning

The pointer is dangling because it points at a temporary instance which was destroyed.

Warning
The pointer is dangling because it points at a temporary instance which was destroyed.
Show more details

Check warning on line 198 in src/osp/Resource/load_tinygltf.cpp

Code scanning

gsl::span or std::string_view created from a temporary will be invalid when the temporary is invalidated (gsl.view).

Warning
gsl::span or std::string_view created from a temporary will be invalid when the temporary is invalidated (gsl.view).
Show more details

Reply:
yep, this is a catastrophic error. Originally intended to keep the importer around, but that isn't the case anymore.